### PR TITLE
WIP Subscription CatalogSource Status

### PR DESCRIPTION
**Pull request by njhale**
Part 1 of 2 in a pair of PRs aimed at surfacing better Subscription status.

### CODE REVIEW

File: pkg/controller/operators/catalog/operator.go [hunk 1]
```diff
@@ -157,18 +173,33 @@ func NewOperator(kubeconfigPath string, logger *logrus.Logger, wakeupInterval ti
...
 		nil,
-		"subscription",
+		subQueueName,
 		metrics.NewMetricsSubscription(op.client),
 		logger,
 	)
-	op.subQueue = subscriptionQueue
-	for _, informer := range subscriptionQueueInformers {
+
+	// Register the QueueInformer for SubscriptionStatus.CatalogStatus
```
Comment (ecordell) [inline pkg/controller/operators/catalog/operator.go]: nit - this comment doesn't seem right

File: pkg/controller/operators/catalog/operator_test.go [hunk 2]
```diff
@@ -464,62 +463,67 @@ func fakeConfigMapData() map[string]string {
 	return data
 }
 
-// NewFakeOperator creates a new operator using fake clients
-func NewFakeOperator(clientObjs []runtime.Object, k8sObjs []runtime.Object, extObjs []runtime.Object, regObjs []runtime.Object, namespace string, stopCh <-chan struct{}) (*Operator, []cache.InformerSynced, error) {
+// NewFakeOperator creates a new operator using fake clients.
+func NewFakeOperator(clientObjs []runtime.Object, k8sObjs []runtime.Object, extObjs []runtime.Object, regObjs []runtime.Object, operatorNamespace string, namespaces []string, stopCh <-chan struct{}) (*Operator, error) {
```
Comment (ecordell) [inline pkg/controller/operators/catalog/operator_test.go]: the `namespace` to `operatorNamespace` change is confusing - in other parts of the tests we use `namespace` to refer to the namespace that OLM is running in and `operatorNamespace` to refer to the namespace that OLM is watching. 

i.e. if we change it here we should change it everywhere to be consistent

File: pkg/api/apis/operators/referencer.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+	v1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/references"
+)
+
+var _ references.ObjectReferencerFunc = ObjectReferenceFor
+
+// ObjectReferenceFor generates an ObjectReference for the given resource if it's provided by the operators.coreos.com API group.
+func ObjectReferenceFor(obj interface{}) (*corev1.ObjectReference, error) {
```
[the remark this answers is not in the document]
Comment (njhale) [reply pkg/api/apis/operators/referencer.go]: I had assumed a util like this existed, but my searching over `kubernetes/kubernetes` didn't pick it up. Is there a canonical reference for useful kube utils that I'm missing???

**Comment by njhale** [comment]
Closing out in favor of starting with #820

**Pull request closed without merging**